client/my-sites/customer-home/cards/features/domain-upsell/test/index.jsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import nock from 'nock';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import DomainUpsell from '../';
+
+jest.mock( 'calypso/state/ui/selectors', () => {
+	return {
+		getSelectedSite: () => {
+			return {
+				ID: 1,
+				plan: 'free',
+			};
+		},
+		getSelectedSiteSlug: () => 'example.wordpress.com',
+	};
+} );
+
+jest.mock( 'calypso/state/sites/domains/selectors', () => {
+	return {
+		getDomainsBySiteId: () => {
+			return [ 'example.wordpress.com' ];
+		},
+	};
+} );
+
+jest.mock( '@automattic/calypso-products', () => {
+	return {
+		isFreePlanProduct: () => true,
+	};
+} );
+
+jest.mock( '@automattic/domain-picker/src', () => {
+	return {
+		useDomainSuggestions: () => {
+			return {
+				allDomainSuggestions: [
+					{
+						is_free: false,
+						product_slug: 'mydomain.com',
+					},
+				],
+			};
+		},
+	};
+} );
+
+jest.mock( 'calypso/state/current-user/selectors', () => ( {
+	getCurrentUserId: jest.fn( () => 12 ),
+	isUserLoggedIn: jest.fn( () => true ),
+	isCurrentUserEmailVerified: jest.fn( () => true ),
+} ) );
+
+let pageLink = '';
+jest.mock( 'page', () => ( link ) => ( pageLink = link ) );
+
+describe( 'index', () => {
+	test( 'Should show H3 content for the Home domain upsell and test search domain button link', async () => {
+		const initialState = {};
+		const mockStore = configureStore();
+		const store = mockStore( initialState );
+
+		const { container } = render(
+			<Provider store={ store }>
+				<DomainUpsell />
+			</Provider>
+		);
+
+		const notice = container.querySelector( 'h3' );
+		expect( notice ).toBeVisible();
+		expect( notice ).toHaveTextContent( 'Own your online identity with a custom domain' );
+
+		const links = [].slice.call( container.querySelectorAll( 'a' ) );
+		expect(
+			links.some( ( link ) =>
+				link.href.endsWith( '/domains/add/example.wordpress.com?domainAndPlanPackage=true' )
+			)
+		).toBeTruthy();
```

Comment:
**Suggestion (non-blocking)**: If possible, I always try to use the more "what the user sees" centric queries provided by React Testing Library! Like `getByRole`.  [Context](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time) and [API](https://testing-library.com/docs/queries/about/#priority).

So, for example...
```javascript
expect( screen.getByRole( 'heading', { name: 'Own your online identity with a custom domain' } ) ).toBeInTheDocument();
```

I like those because they...
1. Test your component more like a real user
2. End up being more resilient over time! (Like, if you ended up swapping out the heading level.)

Not required at all, but just a suggestion! 😄